content/en/docs/tasks/access-kubernetes-api/configure-aggregation-layer.md

```diff
@@ -22,6 +22,100 @@ Configuring the [aggregation layer](/docs/concepts/extend-kubernetes/api-extensi
+
+The apiserver now is prepared to send the request to the extension server.
+
+### Apiserver Proxy the Request
+
+The apiserver now will send, or proxy, the request to the extension server that registered to handle the request. In order to do so, it needs to know several things:
+
+1. How should the apiserver authenticate to the extension server, informing the extension server that the request, which comes over the network, is coming from a valid apiserver?
+2. How should the apiserver inform the extension server of the username and group for which the original request was authenticated?
+
+In order to provide for these two, you must configure the apiserver using several flags.
+
+#### Apiserver Client Authentication
+
+The Kubernetes apiserver authenticates to the extension server using a client certificate. You must provide the following to the Kubernetes apiserver upon startup, using the provided flags:
+
+* private key file via `--proxy-client-key-file`
+* signed client certificate file via `--proxy-client-cert-file`
+* certificate of the CA that signed the client certificate file via `--requestheader-client-ca-file`
+* valid Common Names (CN) in the signed client certificate via `--requestheader-allowed-names`
+
+The Kubernetes apiserver will use the files indicated by `--proxy-client-*-file` to authenticate to the extension server. In order for the request to be considered valid by a compliant extension server, the following conditions must be met:
+
+1. The connection must be made using a client certificate that is signed by the CA whose certificate is in `--requestedheader-client-ca-file`.
+2. The connection must be made using a client certificate whose CN is one of those listed in `--requestheader-allowed-names`. **Note:** You can set this option to blank as `--requestheader-allowed-names=""`. This will indicate to an extension server that _any_ CN is acceptable.
+
+When started with these options, the apiserver will:
+
+1. Use them to authenticate to the extension server.
+2. Create a configmap in the `kube-system` namespace called `extension-apiserver-authentication`, in which it will place the CA certificate and the allowed CNs. These in turn can be retrieved by extension servers to validate requests.
+
+Note that the same client certificate is used by the apiserver to authenticate against _all_ extension servers. It does not create a client certificate per extension server, but rather a single one to authenticate as the apiserver. This same one is reused for all extension server requests. 
+
+#### Original Request Username and Group
+
+When the apiserver proxies the request to the extension server, it informs the extension server of the username and group with which the original request successfully authenticated. It provides these in http headers of its proxied request. You must inform the apiserver of the names of the headers to be used.
```

Comment:
might also be worth noting that the header names are also placed in the config map for consumption by the extension API server